### PR TITLE
making row groups collapsable

### PR DESCRIPTION
**Pull request by ttobollik**
## Summary

- [Ticket](#)
- [Design](#)
- [Storybook](#)

### General

- [ ] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [ ] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--docs).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [x] The changes do not contain any breaking changes.
- [ ] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--docs).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--docs).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--docs).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--docs).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--docs)

### CODE REVIEW

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 1]
```diff
@@ -13,13 +13,40 @@ export class SwirlTableRowGroup {
 
   @Prop() label!: string;
   @Prop() tooltip?: string;
+  @Prop() collapsable?: boolean = false;
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:16]: Issue: Typo (collapsible)

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 2]
```diff
@@ -13,13 +13,40 @@ export class SwirlTableRowGroup {
...
+
+  @State() isVisible: boolean = true;
+
+  private toggleShowItems = () => {
+    const rowsContainer = this.el.shadowRoot.querySelector(
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:21]: Suggestion: Use the JSX ref attribute to get the element reference. https://stenciljs.com/docs/templating-jsx#getting-a-reference-to-a-dom-element

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 3]
```diff
@@ -13,13 +13,40 @@ export class SwirlTableRowGroup {
...
+      ".rows-container"
+    ) as HTMLElement;
+    if (this.isVisible) {
+      rowsContainer.style.height = "0";
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:25]: Issue: We need to hide the collapsed contents from the accessibility tree. You could use `display: none;` or `aria-hidden="true"` for example.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 4]
```diff
@@ -13,13 +13,40 @@ export class SwirlTableRowGroup {
...
+    } else {
+      rowsContainer.style.height = "auto";
+      const height = rowsContainer.clientHeight + "px";
+      rowsContainer.style.height = "0";
+      setTimeout(() => (rowsContainer.style.height = height), 0);
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:30]: Suggestion: There is a neat trick to be able to animate to auto height with CSS only. Maybe this works here.
https://css-tricks.com/css-grid-can-do-auto-height-transitions/

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 5]
```diff
@@ -43,7 +70,9 @@ export class SwirlTableRowGroup {
             )}
           </span>
         </div>
-        <slot></slot>
+        <div class={{ "rows-container": true, "is-hidden": !this.isVisible }}>
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:73]: Issue: For consistency BEM-style class names would be nice. So:
`rows-container` -> `table-row-group__rows-container` 
and
`is-hidden` -> `table-row-group__rows-container--is-hidden`

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 4 above)
Comment (Sqrrl) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:30]: Issue: The first collapsing isn't animated.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 6]
```diff
@@ -13,13 +13,40 @@ export class SwirlTableRowGroup {
...
     const rowspan = this.el.querySelectorAll("swirl-table-row").length;
+    const Icon = this.isVisible
+      ? "swirl-icon-expand-less"
+      : "swirl-icon-expand-more";
 
     return (
       <Host class="table-row-group" role="rowgroup">
         <div class="table-row-group__header-row" role="row">
+          {this.collapsable && (
+            <Icon
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:45]: Issue: This needs to be a button with a label. The label should probably be adjustable via a prop.

Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:45]: Suggestion: Would be great if you could implement the Disclosure pattern here. It's pretty straight forward. Just an aria-expanded on the button. https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/

**Review by Sqrrl — CHANGES_REQUESTED** [review]

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 1 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: done

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 2 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: Yes, that works. Cool, thanks!

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 4 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: I tried multiple times to use the css as you described (I left my approach in comments) But i could not get it to work. Either no row or part of a row disappeared, but not more. 
I added a lot of animation scheduling and the first collapse should be animated now. :)

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 6 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: done :)

Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: I see! I added it. Cool, thank you for the resources!!

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 5 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: Good point. Your review is encouraging me to develop more in swirl. I am learning a lot.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 7]
```diff
@@ -13,13 +13,80 @@ export class SwirlTableRowGroup {
...
       <Host class="table-row-group" role="rowgroup">
-        <div class="table-row-group__header-row" role="row">
+        <div
+          class="table-row-group__header-row"
+          role="row"
+          ref={(el) => (this.headerRowElement = el as HTMLElement)}
+        >
+          {this.collapsible && (
+            <swirl-button
+              tabIndex={0}
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:77]: Issue: Not needed. The swirl-button has a native button element inside, which already has a tab stop.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 8]
```diff
@@ -13,13 +13,80 @@ export class SwirlTableRowGroup {
...
+              class="table-row-group__collapse-icon"
+              hideLabel
+              icon={iconType}
+              label={this.collapseButtonLabel}
+              variant="plain"
+              onClick={this.toggleShowItems}
+              onKeyPress={this.handleKeyPress}
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:84]: Issue: Should not be necessary. The onClick should trigger for Enter and Space keys.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 9]
```diff
@@ -13,13 +13,80 @@ export class SwirlTableRowGroup {
...
+              iconPosition="end"
+              role="button"
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:86]: Issue: Remove. The swirl-button wraps a native button.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx [hunk 10]
```diff
@@ -13,13 +13,80 @@ export class SwirlTableRowGroup {
...
+              aria-expanded={this.isVisible}
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx:87]: Issue: The aria-expanded  needs to be added to the native button element inside the swirl-button. You can do that via the `swirl-aria-expanded` attribute of the swirl-button. I know, the swirl-button is a bit confusing. ;)

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 7 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: ! that makes a lot of sense.

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 9 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: Understood :)

File: packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx (hunk 10 above)
Comment (ttobollik) [reply packages/swirl-components/src/components/swirl-table-row-group/swirl-table-row-group.tsx]: Ah! I see, thank you for that :)

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**